documentation/glossary.md

```diff
@@ -119,6 +119,10 @@ With [Filecoin Pin](#filecoin-pin), the Piece is the [CAR](#car) file itself; an
 
 PieceCID, or "CommP" (Commitment of Piece), is a specific form of [CID](#cid) used in Filecoin to commit Merkle proofs of large _pieces_ of data on chain. A PieceCID includes a digest of the contiguous bytes, with no special handling of any internal format or packing (including CAR formats containing IPFS data). It uses a modified form of SHA2-256 internally, and further details can be found in [FRC-0069](https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0069.md). PieceCID is a variant of CID specifically for use in Filecoin's proof system, and will differ from the CIDs used in IPFS. When presented in standard base32 format, it will begin with the characters `bafkzcib` and be between 64 and 65 characters long.
 
+## Streaming Uploads
+
+[Filecoin Pin](#filecoin-pin) streams CAR data to [Service Providers](#service-provider) via [Synapse](#synapse), so files are not buffered fully in memory during upload. The maximum supported piece size is bounded by the Synapse SDK and SP configuration (tracked in https://github.com/FilOzone/synapse-sdk/issues/110).
```

Comment:
needs tweaking as well, in the browser we buffer up to 1GiB (1016MiB)

in fact, I don't think we even have early limits on it, so potentially someone could request an `add` of >1GiB and it'd just buffer that up and not error until we eventually try to `upload()` it; which could be nasty for a browser environment. We could add some checks in there if we want to be safe now (keeping in mind that final CAR size is going to be larger than input size, but even a check of inputsize<`MAX_UPLOAD_SIZE` would help). But also we could just put it off until https://github.com/filecoin-project/filecoin-pin/issues/288 which will give us the check real-time as we're making the CAR from the input and hit that limit.